packages/edit-site/src/components/global-styles/font-library-modal/upload-fonts.js

```diff
@@ -45,29 +45,50 @@ function UploadFonts() {
 	 * @param {Array} files The files to be filtered
 	 * @return {void}
 	 */
-	const handleFilesUpload = ( files ) => {
+	const handleFilesUpload = async ( files ) => {
 		setNotice( null );
 		setIsUploading( true );
 		const uniqueFilenames = new Set();
 		const selectedFiles = [ ...files ];
-		const allowedFiles = selectedFiles.filter( ( file ) => {
+		let hasInvalidFiles = false;
+
+		// Use map to create a promise for each file check, then filter with Promise.all
+		const checkFilesPromises = selectedFiles.map( async ( file ) => {
```

Comment:
I think it could probably be made a little simpler.

```javascript
const allowedFiles = (
	await Promise.all(
		selectedFiles.map( async ( file ) => {
			// ...
		} )
	)
).filter( Boolean );
```

Reply:
I agree that it's an elegant way to use `.filter`, but I think that since we are explicitly trying to filter out `null` files, it's clearer for the reader to add the specific condition instead of a blanket `Boolean`.